### DISCUSSION TITLE
windows-2019 hosted agent: creating Docker container from tests fails (no Docker Environment found)

### DISCUSSION THREAD

**Discussion by LucasReehorst** [actions/virtual-environments#3937] [Installed software]
When we run our Azure DevOps pipeline on the windows-2019 Microsoft-hosted agent, we find that the tests we run as part of a Maven 3 task that requires a Docker installation to be present fail (we are trying to run tests that use https://www.testcontainers.org/ to provide us with a ms sql server). 

When we run the tests locally the tests succeed and we see the following log statements: 

INFO  o.testcontainers.DockerClientFactory : Checking the system...
INFO  o.testcontainers.DockerClientFactory : ✔︎ Docker server version should be at least 1.6.0
INFO  o.testcontainers.DockerClientFactory : ✔︎ Docker environment should have more than 2GB free disk space
INFO  ?.m.com/mssql/server:2017-CU12] : Creating container for image: mcr.microsoft.com/mssql/server:2017-CU12

When we run it on the windows-2019 agent, the tests fail and we see this: 
- java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration

Docker 20.10.6 is listed in the Installed Software for the windows-2019 agent.
Can anyone help me figure out what the problem might be? (If anyone could point me to a more appropriate place to ask this question, that would be appreciated too.)

**Comment by LucasReehorst** [comment]
Discovered this issue does not occur when we use the ubuntu-20.04 hosted-agent, which I think solves the issue for us. 
I would still appreciate it if anyone could help me understand why our setup does not work on the windows-2019 agent (when I run the tests locally on Windows 10 everything works fine).

**Reply by miketimofeev** [reply]
@LucasReehorst the mcr.microsoft.com/mssql/server:2017-CU12 image exists only for Linux but in the windows-2019 VM image we only have windows containers support. If you try to `docker pull mcr.microsoft.com/mssql/server:2017-CU12` you'll get the following error on windows-2019:
```
2017-CU12: Pulling from mssql/server
image operating system "linux" cannot be used on this platform
```
